packages/amplify-util-mock/src/__e2e__/utils/index.ts

```diff
@@ -0,0 +1,96 @@
+import { AmplifyAppSyncSimulator } from 'amplify-appsync-simulator';
+import { ensureDynamoDBTables, configureDDBDataSource } from '../../utils/ddb-utils';
+import { processAppSyncResources } from '../../CFNParser';
+import * as dynamoEmulator from 'amplify-dynamodb-simulator';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { getFunctionDetails } from './lambda-helper';
+import { invoke } from '../../utils/lambda/invoke';
+
+export async function launchDDBLocal() {
+    let dbPath
+    while(true) {
+        dbPath = path.join(
+            '/tmp',
+            `amplify-cli-emulator-dynamodb-${Math.floor(Math.random() * 100)}`
+        );
+        if(!fs.existsSync(dbPath))
+            break;
+    }
+
+    fs.ensureDirSync(dbPath);
+    const emulator = await dynamoEmulator.launch({
+        dbPath,
+        port: null
+    });
+    const client = await dynamoEmulator.getClient(emulator);
+    return { emulator, dbPath, client };
+}
+
+export async function deploy(transformerOutput: any, client = null) {
+    const stacks = Object.values(transformerOutput.stacks).reduce(
+        (prev, stack: any) => {
+            return { ...prev, ...stack.Resources };
+        },
+        { ...transformerOutput.rootStack.Resources }
+    );
+
+    let config:any = processAppSyncResources(stacks, transformerOutput);
+    if(config.appSync.authenticationType == "API_KEY" && !config.appSync.apiKey) {
+        // transformer generates API Key only if AuthTransformer is included
+        config.appSync.apiKey = 'da-fake-api-key';
+    }
+    if(client) {
+        await ensureDynamoDBTables(client, config);
+        config = configureDDBDataSource(config, client.config);
+    }
+    configureLambdaDataSource(config);
+    const simulator = await runAppSyncSimulator(config);
+    return {simulator, config};
+}
+async function configureLambdaDataSource(config) {
+    config.dataSources
+        .filter(d => d.type === 'AWS_LAMBDA')
+        .forEach(d => {
+            const arn = d.LambdaFunctionArn;
+            const arnParts = arn.split(':');
+            let functionName = arnParts[arnParts.length - 1];
```

Comment:
I can't see the env name handlig here what is present in `api.ts`. Don't we need the same functionality here? (if yes, we should dedup the code and move to some utility function if possible).

Reply:
When running test we don't provision resource like how we do in a project. Here the functions are put in a fixed folder and it does not have env prepended. These are not true e2e but more of running output from transformer output without creating a project